contracts/src/L2/predeploys/L1GasPriceOracle.sol

```diff
@@ -129,4 +210,57 @@ contract L1GasPriceOracle is OwnableBase, IL1GasPriceOracle {
         whitelist = IWhitelist(_newWhitelist);
         emit UpdateWhitelist(_oldWhitelist, _newWhitelist);
     }
+
+    /// @notice Enable the Curie fork (callable by contract owner).
+    ///
+    /// @dev Since this is a predeploy contract, we will directly set the slot while hard fork
+    /// to avoid external owner operations.
+    /// The reason that we keep this function is for easy unit testing.
+    function enableCurie() external onlyOwner {
+        if (isCurie) revert ErrAlreadyInCurieFork();
+        isCurie = true;
+    }
+
+    /**********************
+     * Internal Functions *
+     **********************/
+
+    /// @dev Internal function to computes the amount of L1 gas used for a transaction before Curie fork.
+    ///   The `_data` is the RLP-encoded transaction with signature. And we also reserve additional
+    ///   4 bytes in the non-zero bytes to store the number of bytes in the RLP-encoded transaction.
+    /// @param _data Signed fully RLP-encoded transaction to get the L1 gas for.
+    /// @return Amount of L1 gas used to publish the transaction.
+    function _getL1GasUsedBeforeCurie(bytes memory _data) private view returns (uint256) {
+        uint256 _total = 0;
+        uint256 _length = _data.length;
+        unchecked {
+            for (uint256 i = 0; i < _length; i++) {
+                if (_data[i] == 0) {
+                    _total += 4;
+                } else {
+                    _total += 16;
+                }
+            }
+            return _total + overhead + (4 * 16);
+        }
+    }
+
+    /// @dev Internal function to compute the L1 portion of the fee based on the size of the rlp encoded input
+    ///   transaction, the current L1 base fee, and the various dynamic parameters, before Curie fork.
+    /// @param _data Signed fully RLP-encoded transaction to get the L1 fee for.
+    /// @return L1 fee that should be paid for the tx
+    function _getL1FeeBeforeCurie(bytes memory _data) private view returns (uint256) {
+        uint256 _l1GasUsed = _getL1GasUsedBeforeCurie(_data);
+        uint256 _l1Fee = _l1GasUsed * l1BaseFee;
+        return (_l1Fee * scalar) / PRECISION;
+    }
+
+    /// @dev Internal function to compute the L1 portion of the fee based on the size of the rlp encoded input
+    ///   transaction, the current L1 base fee, and the various dynamic parameters, after Curie fork.
+    /// @param _data Signed fully RLP-encoded transaction to get the L1 fee for.
+    /// @return L1 fee that should be paid for the tx
+    function _getL1FeeCurie(bytes memory _data) private view returns (uint256) {
+        // We have bounded the value of `commitScalar` and `blobScalar`, the whole expression won't overflow.
+        return (commitScalar * l1BaseFee + blobScalar * _data.length * l1BlobBaseFee) / PRECISION;
```

Comment:
this `_data.length` refer to tx call data length? the tx calls this contract method

Reply:
Signed fully RLP-encoded transaction to get the L1 fee for.